R/exercise.R

```diff
@@ -327,34 +327,32 @@ evaluate_exercise <- function(
     eval(parse(text = exercise$global_setup), envir = envir)
   }
 
-  # Check if user code is parsable
-  parse_results <- tryCatch(
-    parse(text = exercise$code),
-    error = function(e) exercise_result_parse_error(e, exercise)
-  )
+  # Check if user code has unfilled blanks; if it does, early return
+  exercise_blanks <- exercise$options$exercise.blanks %||%
+    knitr::opts_chunk$get("exercise.blanks") %||%
+    "_{3,}"
 
-  if (is_exercise_result(parse_results)) {return(parse_results)}
+  if (isTruthy(exercise_blanks)) {
```

Comment:
Super nit, but let's prefix this so it's clear we didn't come up with our own version.

```suggestion
  if (shiny::isTruthy(exercise_blanks)) {
```